### DISCUSSION TITLE
Welcome to smarty Discussions!

### DISCUSSION THREAD

**Discussion by wisskid** [smarty-php/smarty#624] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by scottchiefbaker** [comment]
This is a great idea. I've been using Smarty for a long time, and I love it.

Looking forward to submitting documentation updates, and testing out PHP 8.x support.

Thanks for the great project.

**Comment by Billjay65** [comment]
Hi, 
I have been using smarty 3 and now I am trying to upgrade to smarty 4 since I updated my php to version 8.0. Unfortunately, I have not succeeded. 


I am trying to upgrade my PHP smarty template system from smarty 3 to smarty-4.1.0. I have followed the recommended procedure and setup all of the required directories. However, when I try to run my application I keep getting this error message:

ERRNO: 2
TEXT: Attempt to read property "_codeFrame" on null
LOCATION: C:\Users\JAY.V\Desktop\campbroot\campb\libs\smarty\sysplugins\smarty_internal_templatecompilerbase.php, line 392, at July 14, 2022, 1:22 am
Showing backtrace:
Smarty_Internal_TemplateCompilerBase.compileTemplate(Object: Smarty_Internal_Template) # line  184, file: C:\Users\JAY.V\Desktop\campbroot\campb\libs\smarty\sysplugins\smarty_template_compiled.php
Smarty_Template_Compiled.compileTemplateSource(Object: Smarty_Internal_Template) # line  141, file: C:\Users\JAY.V\Desktop\campbroot\campb\libs\smarty\sysplugins\smarty_template_compiled.php
Smarty_Template_Compiled.process(Object: Smarty_Internal_Template) # line  105, file: C:\Users\JAY.V\Desktop\campbroot\campb\libs\smarty\sysplugins\smarty_template_compiled.php
Smarty_Template_Compiled.render(Object: Smarty_Internal_Template) # line  216, file: C:\Users\JAY.V\Desktop\campbroot\campb\libs\smarty\sysplugins\smarty_internal_template.php
Smarty_Internal_Template.render(false, "1") # line  238, file: C:\Users\JAY.V\Desktop\campbroot\campb\libs\smarty\sysplugins\smarty_internal_templatebase.php
Smarty_Internal_TemplateBase._execute(Object: Smarty_Internal_Template, null, null, null, "1") # line  134, file: C:\Users\JAY.V\Desktop\campbroot\campb\libs\smarty\sysplugins\smarty_internal_templatebase.php
Smarty_Internal_TemplateBase.display("store_front.tpl") # line   80, file: C:\Users\JAY.V\Desktop\campbroot\campb\index.php

When I run the smarty installation test, everything seems to be in place as shown below. I don't know where the error is coming from.

Smarty Installation test...
Testing template directory...
C:\Users\JAY.V\Desktop\campbroot\campb\presentation\templates is OK.
Testing compile directory...
C:\Users\JAY.V\Desktop\campbroot\campb\presentation\templates_c is OK.
Testing plugins directory...
C:\Users\JAY.V\Desktop\campbroot\campb\libs\smarty\plugins is OK.
C:\Users\JAY.V\Desktop\campbroot\campb\presentation\smarty_plugins is OK.
Testing cache directory...
C:\Users\JAY.V\Desktop\campbroot\campb\libs\smarty\cache is OK.
Testing configs directory...
C:\Users\JAY.V\Desktop\campbroot\campb\include\configs\ is OK.
Testing sysplugin files...
... OK
Testing plugin files...
... OK
Tests complete.

Here is the code that calls for smarty in index.php

$application->testInstall();
   
  // Display the page
  $application->display('store_front.tpl');

I have searched all over the internet for a solution to no avail. Please help me out.

**Reply by wisskid** [reply]
Why are you calling `Smarty_Internal_TemplateBase.display` in index.php? `Smarty_Internal_TemplateBase` is an internal class. Can you show me the contents of `index.php`?
